src/Analyzer.JsonRuleEngine.UnitTests/InequalityOperatorTests.cs

```diff
@@ -145,9 +145,9 @@ private void CompareObjects(object left, object right, bool greater = false, boo
             var leftToken = TestUtilities.ToJToken(left);
             var rightToken = TestUtilities.ToJToken(right);
 
-            var inequalityOperator = new InequalityOperator(leftToken, greater: greater, orEquals: orEquals);
+            var inequalityOperator = new InequalityOperator(rightToken, greater: greater, orEquals: orEquals);
```

Comment:
I don't understand this change, it seems like you're flipping it around so I think I am missing something? Why is the value given to the constructor operating on as the right hand operator for the comparison? I feel like it makes more sense how it already is?

Reply:
I confused the value in the template that the operator checks against with the value in the rule, here's an example: https://github.com/Azure/template-analyzer/issues/159

Reply:
I thought this was confusing at first too, but now I think this is actually a better representation.

If a rule is written like:
```
{
    "path": "property.value"
    "lessThan": 50
}
```
... the concept being expressed is `'property.value' < 50`, in which case the `InequalityOperator` should actually be initialized with the right token.